spring-boot/admin/src/main/java/eu/coatrack/admin/service/OauthUserAccountManagement.java

```diff
@@ -0,0 +1,127 @@
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import eu.coatrack.config.github.GithubEmail;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.oauth2.client.OAuth2AuthorizedClient;
+import org.springframework.security.oauth2.client.OAuth2AuthorizedClientService;
+import org.springframework.security.oauth2.client.authentication.OAuth2AuthenticationToken;
+import org.springframework.security.oauth2.core.user.OAuth2User;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+
+import java.util.List;
+
+@Service
+public class OauthUserAccountManagement {
+
+    private static final String GITHUB_API_EMAIL = "https://api.github.com/user/emails";
+
+    @Autowired
+    private OAuth2AuthorizedClientService clientService;
+
+    private OAuth2User getAttributesFromAuthentication() {
+        OAuth2User loggedInUser = (OAuth2User) SecurityContextHolder
+                .getContext()
+                .getAuthentication()
+                .getPrincipal();
+        return loggedInUser;
+    }
+
+    public String getLoginNameFromLoggedInUser() {
+        return getAttributesFromAuthentication().getAttribute("login");
+    }
+
+    public String getNameFromLoggedInUser() {
+        return getAttributesFromAuthentication().getAttribute("name");
+    }
+
+    public String getCompanyFromLoggedInUser() {
+        return getAttributesFromAuthentication().getAttribute("company");
+    }
+
+    public String getEmailFromLoggedInUser() throws JsonProcessingException {
+        String email = getAttributesFromAuthentication().getAttribute("email");
+
+        // If the email is defined as private on GitHub (The Oauth2 will retrieve the
+        // attribute email as null),
+        // so the only way to retrieve is to make request with the Oauth Token
+        if (email == null || email.isEmpty()) {
```

Comment:
Maybe it would be a better solution to make a function out of that like:
```java
if (isGitHubMailAddressSetToPrivateAndThereforeNotReadable(email))
```
So you wouldn't need any comments.

Reply:
Done